File: OrderMan/dev/modules/customer/Customer.ts
export class Customer {
  id: number;
  name: string;
  address: string;
  phone: string;
  bankAccount?: string;
  createdAt: Date;
  updatedAt: Date;
  orders: number[];

  constructor(
    id: number,
    name: string,
    address: string,
    phone: string,
    bankAccount?: string,
    createdAt?: Date,
    updatedAt?: Date
  ) {
    this.id = id;
    this.name = name;
    this.address = address;
    this.phone = phone;
    this.bankAccount = bankAccount;
    this.createdAt = createdAt || new Date();
    this.updatedAt = updatedAt || new Date();
    this.orders = [];
  }
}

export default Customer;

File: method_pipeline/generated_test/DeepSeek/chatdev-040726-QA1/sdk_chatdev_20260704154239_20260704154239/code_workspace/oms/app/models/customer.py
"""
Customer ORM model.
"""
import uuid
from datetime import datetime, timezone

from sqlalchemy import Column, String, DateTime, Text
from sqlalchemy.orm import relationship

from app.database import Base


class Customer(Base):
    __tablename__ = "customers"

    id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    name = Column(String(255), nullable=False, index=True)
    address = Column(Text, nullable=False)
    phone = Column(String(50), nullable=False)
    banking_details = Column(Text, nullable=False)
    role = Column(String(50), nullable=False, default="customer")
    created_at = Column(DateTime, default=lambda: datetime.now(timezone.utc), nullable=False)
    updated_at = Column(DateTime, default=lambda: datetime.now(timezone.utc), onupdate=lambda: datetime.now(timezone.utc), nullable=False)

    orders = relationship("Order", back_populates="customer", cascade="all, delete-orphan")

    def __repr__(self) -> str:
        return f"<Customer {self.id} {self.name}>"
